include/boost/geometry/algorithms/is_empty.hpp

```diff
@@ -62,10 +62,10 @@ class polygon_is_empty
     template <typename InteriorRings>
     static inline bool check_interior_rings(InteriorRings const& interior_rings)
     {
-        return check_iterator_range
-            <
-                range_is_empty, true // allow empty range
-            >::apply(boost::begin(interior_rings), boost::end(interior_rings));
+        return boost::end(interior_rings) == std::find_if(
+            boost::begin(interior_rings), 
+            boost::end(interior_rings),
+            []( auto const& range ){ return ! range_is_empty::apply(range); });
```

Comment:
> This could probably be replaced with:
> 
> ```
> []( auto const& range ){ return ! boost::empty(range); }
> ```

+1

> You could also check if `range_is_empty` is needed because it's possible that it's not used anywhere else.

It is used as default policy to `multi_is_empty`